src/widgets/scroll/scroll_bar.rs

```diff
@@ -0,0 +1,303 @@
+use crate::core::{
+    render_command::RenderCommand,
+    rsx,
+    styles::{PositionType, Style, Units},
+    use_state, widget, Bound, EventType, MutableBound, OnEvent, WidgetProps,
+};
+use kayak_core::layout_cache::Rect;
+use kayak_core::styles::{Corner, Edge};
+use kayak_core::Color;
+
+use crate::widgets::Background;
+
+use super::{map_range, ScrollContext};
+
+/// Props used by the [`ScrollBar`] widget
+#[derive(WidgetProps, Default, Debug, PartialEq, Clone)]
+pub struct ScrollBarProps {
+    /// If true, disables the ability to drag
+    pub disabled: bool,
+    /// If true, displays a horizontal scrollbar instead of a vertical one
+    pub horizontal: bool,
+    /// The thickness of the scrollbar in pixels
+    pub thickness: f32,
+    /// The color of the scrollbar thumb
+    pub thumb_color: Option<Color>,
+    /// The styles of the scrollbar thumb
+    pub thumb_styles: Option<Style>,
+    /// The color of the scrollbar track
+    pub track_color: Option<Color>,
+    /// The styles of the scrollbar track
+    pub track_styles: Option<Style>,
+    #[prop_field(Styles)]
+    styles: Option<Style>,
+}
+
+#[widget]
+/// A widget that displays the current scroll progress within a [`ScrollBox`](crate::ScrollBox) widget
+///
+/// This widget consists of two main components:
+///
+/// ### The Thumb
+///
+/// This is the actual indicator for scroll progress. It also allows you to control the
+/// scroll offset by dragging it around.
+///
+/// ### The Track
+///
+/// This is the track along which the thumb moves. Clicking anywhere in the track will move
+/// the thumb to the clicked position and update the scroll offset.
+///
+/// # Props
+///
+/// __Type:__ [`ScrollBarProps`]
+///
+/// | Common Prop | Accepted |
+/// | :---------: | :------: |
+/// | `children`  | ❌        |
+/// | `styles`    | ✅        |
+/// | `on_event`  | ❌        |
+/// | `focusable` | ❌        |
+///
+/// # Panics
+///
+/// This widget will panic when used outside the context of a [`ScrollContext`], which
+/// is automatically provided by [`ScrollBox`](crate::ScrollBox).
+pub fn ScrollBar(props: ScrollBarProps) {
+    // === Scroll === //
+    let scroll_ctx = context.create_consumer::<ScrollContext>().unwrap();
```

Comment:
Yeah the goal was to make it so we could calculate the content layout in `ScrollContent` and update the size info on `ScrollContext`. I think we need a better way of getting that info (maybe with a `LayoutEvent` event type?).

It was also supposed to allow nested `ScrollBox` widgets, but I just tested it and it seems to not work like I thought it would :(

In fact, if I wrap my content in an `Element` it just flat out doesn't work. I think maybe it sizes to fit the parent rather than its children? 🤔

Reply:
I think #86 will fix this issue